test-data/unit/check-union-or-syntax.test

```diff
@@ -251,3 +251,65 @@ foo: ReadableBuffer
 [file was_mmap.py]
 from was_builtins import *
 class mmap: ...
+
+[case testUnionOrSyntaxWithQuotedOperandsNotAllowed]
+# flags: --python-version 3.10
+from typing import Union, assert_type
+
+x1: "int" | str          # E: X | Y syntax for unions cannot use quoted operands; use quotes around the entire expression instead
+x2: int | "str"          # E: X | Y syntax for unions cannot use quoted operands; use quotes around the entire expression instead
+x3: int | str | "bytes"  # E: X | Y syntax for unions cannot use quoted operands; use quotes around the entire expression instead
+assert_type(x1, Union[int, str])
+assert_type(x2, Union[int, str])
+assert_type(x3, Union[int, str, bytes])
+
+[case testUnionOrSyntaxWithQuotedOperandsWithTypeVar]
+# flags: --python-version 3.10
+import types
+from typing import TypeVar, Union, assert_type
+from typing_extensions import TypeAlias
+
+T = TypeVar("T")
+
+ok1: TypeAlias =  T | "int"
```

Comment:
Would it be useful to have more tests for things like `"int | str" | T` or `"int | str"`, here and elsewhere?